rust/arrow/src/util/data_gen.rs

```diff
@@ -0,0 +1,347 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Utilities to generate random arrays and batches
+
+use std::{convert::TryFrom, sync::Arc};
+
+use rand::{distributions::uniform::SampleUniform, Rng};
+
+use crate::error::{ArrowError, Result};
+use crate::record_batch::{RecordBatch, RecordBatchOptions};
+use crate::{array::*, datatypes::SchemaRef};
+use crate::{
+    buffer::{Buffer, MutableBuffer},
+    datatypes::*,
+};
+
+use super::{bench_util::*, bit_util, test_util::seedable_rng};
+
+/// Create a random [RecordBatch] from a schema
+pub fn create_random_batch(
+    schema: SchemaRef,
+    size: usize,
+    null_density: f32,
+    true_density: f32,
+) -> Result<RecordBatch> {
+    let columns = schema
+        .fields()
+        .iter()
+        .map(|field| create_random_array(field, size, null_density, true_density))
+        .collect::<Result<Vec<ArrayRef>>>()?;
+
+    RecordBatch::try_new_with_options(
+        schema,
+        columns,
+        &RecordBatchOptions {
+            match_field_names: false,
+        },
+    )
+}
+
+/// Create a random [ArrayRef] from a [DataType] with a length,
+/// null density and true density (for [BooleanArray]).
+pub fn create_random_array(
+    field: &Field,
+    size: usize,
+    null_density: f32,
+    true_density: f32,
+) -> Result<ArrayRef> {
+    // Override null density with 0.0 if the array is non-nullable
+    let null_density = match field.is_nullable() {
+        true => null_density,
+        false => 0.0,
+    };
+    use DataType::*;
+    Ok(match field.data_type() {
+        Null => Arc::new(NullArray::new(size)) as ArrayRef,
+        Boolean => Arc::new(create_boolean_array(size, null_density, true_density)),
+        Int8 => Arc::new(create_primitive_array::<Int8Type>(size, null_density)),
+        Int16 => Arc::new(create_primitive_array::<Int16Type>(size, null_density)),
+        Int32 => Arc::new(create_primitive_array::<Int32Type>(size, null_density)),
+        Int64 => Arc::new(create_primitive_array::<Int64Type>(size, null_density)),
+        UInt8 => Arc::new(create_primitive_array::<UInt8Type>(size, null_density)),
+        UInt16 => Arc::new(create_primitive_array::<UInt16Type>(size, null_density)),
+        UInt32 => Arc::new(create_primitive_array::<UInt32Type>(size, null_density)),
+        UInt64 => Arc::new(create_primitive_array::<UInt64Type>(size, null_density)),
+        Float16 => {
+            return Err(ArrowError::NotYetImplemented(
+                "Float16 is not implememted".to_string(),
```

Comment:
```suggestion
                "Float16 is not implemented".to_string(),
```

Reply:
Sorry, that's how you say it in an African accent 😄